src/dcat-us/dataset-formatter.ts

```diff
@@ -1,163 +1,31 @@
-    id: hubDataset.id, 
-    attributes: hubDataset
-  } as DatasetResource);
-  const { relative: relativePath } = getContentSiteUrls(content, siteModel);
-  const hubLandingPage = siteUrl.startsWith('https://') ? siteUrl + relativePath : `https://${siteUrl}${relativePath}`;
-  const downloadLink = siteUrl.startsWith('https://') ? `${siteUrl}/datasets/${content.identifier}` : `https://${siteUrl}/datasets/${content.identifier}`;
-
-  // AGO links must be generated from Dataset Records
-  const { itemId, layerId } = parseDatasetId(hubDataset.id);
-  let agoLandingPage = `${portalUrl}/home/item.html?id=${itemId}`;
-  if (layerId) {
-    agoLandingPage += `&sublayer=${layerId}`;
-  }
-
-  const { 
-    structuredLicense: { url = null } = {},
-    licenseInfo = ''
-  } = hubDataset;
-
-  // Override hub.js default license value of 'none'
-  if (hubDataset.license === 'none') {
-    hubDataset.license = null;
-  }
-
+export function formatDcatDataset(hubDataset: any, feedTemplate: DcatDatasetTemplate, feedTemplateTransforms: TransformsList) {
   const defaultDataset = {
-    '@type': 'dcat:Dataset',
-    identifier: agoLandingPage,
-    landingPage: hubLandingPage
-  };
-
-  const transforms: TransformsList = {
-    toISO (_key, val) {
-      return new Date(val).toISOString();
-    },
-    toArray (_key, val) {
-      if (!val) return [];
-      else return _.castArray(val);
-    }
+    '@type': 'dcat:Dataset'
   };
 
-  const dcatDataset = Object.assign({}, defaultDataset, adlib(datasetTemplate, hubDataset, transforms));
-
-  // reset uninterpolated, non-editable fields since customers cannot remove them
-  resetUninterpolatedPaths(dcatDataset, nonEditableFieldPaths);
-
-  if (!dcatDataset.license || dcatDataset.license.match(/{{.+}}/g)?.length) {
-    dcatDataset.license = url || licenseInfo || '';
-  }
-
-  if (isPage(hubDataset as IItem) && !hasTags(hubDataset)) {
-    dcatDataset.keyword = ['ArcGIS Hub page'];
-  }
-
-  dcatDataset.distribution = _generateDistributions({
-    hubDataset, 
-    dcatDataset,
-    landingPage: hubLandingPage, 
-    downloadLink
-  });
-
-  const spatial = computeSpatialProperty(datasetTemplate, dcatDataset);
-  if (spatial) {
-    dcatDataset.spatial = spatial;
-
-    // https://project-open-data.cio.gov/v1.1/schema/#theme
-    // allow theme to be overridden
-    if (_.isEmpty(datasetTemplate.theme)) {
-      dcatDataset.theme = ['geospatial'];
-    }
-  } else if (dcatDataset.spatial) {
-    delete dcatDataset.spatial;
+  try {
+    const dcatDataset = Object.assign({}, defaultDataset, adlib(feedTemplate, hubDataset, feedTemplateTransforms));
+    return indent(JSON.stringify({
+      ...dcatDataset,
+      distribution: dcatDataset.distribution && removeUninterpolatedDistributions(dcatDataset.distribution),
+      theme: dcatDataset.spatial && ['geospatial']
+    }, null, '\t'), 2);
+  } catch (err) {
+    throw new RemoteServerError(err?.message || 'Error parsing feed template', null, 400);
```

Comment:
You don't want a "RemoteServerError" as it indicates the error originates from some other upstream server. You'd use this to distinguish an error originating from the hub-search service as opposed to the Koop feeds service.

Reply:
Created Custom error for errors originating at DCAT plugin.